metrics/web/test/features/auth/presentation/widgets/auth_form_test.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/features/auth/presentation/state/auth_store.dart';
+import 'package:metrics/features/auth/presentation/strings/auth_strings.dart';
+import 'package:metrics/features/auth/presentation/widgets/auth_form.dart';
+import 'package:metrics/features/auth/presentation/widgets/auth_input_field.dart';
+import 'package:mockito/mockito.dart';
+import 'package:states_rebuilder/states_rebuilder.dart';
+
+void main() {
+  final emailInputFinder =
+      find.widgetWithText(AuthInputField, AuthStrings.email);
+  final passwordInputFinder =
+      find.widgetWithText(AuthInputField, AuthStrings.password);
+  final submitButtonFinder =
+      find.widgetWithText(RaisedButton, AuthStrings.signIn);
+
+  const testEmail = 'test@email.com';
+  const testPassword = 'testPassword';
+
+  group("AuthForm", () {
+    testWidgets("email input shows error message if value is empty",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+
+      await tester.tap(submitButtonFinder);
+      await tester.pump();
+
+      expect(find.text(AuthStrings.emailIsRequired), findsOneWidget);
+    });
+
+    testWidgets("email input shows error message if value is not a valid email",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+      await tester.enterText(emailInputFinder, 'notAnEmail');
+
+      await tester.tap(submitButtonFinder);
+      await tester.pump();
+
+      expect(find.text(AuthStrings.emailIsInvalid), findsOneWidget);
+    });
+
+    testWidgets("password input shows error message if value is empty",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+
+      await tester.tap(submitButtonFinder);
+      await tester.pump();
+
+      expect(find.text(AuthStrings.passwordIsRequired), findsOneWidget);
+    });
+
+    testWidgets("password input shows error message if value is less then 6",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+
+      await tester.enterText(passwordInputFinder, '12345');
+      await tester.tap(submitButtonFinder);
+      await tester.pump();
+
+      expect(find.text(AuthStrings.passwordMinLength), findsOneWidget);
+    });
+
+    testWidgets("password input shows a value replaced by bullets",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+      await tester.enterText(passwordInputFinder, 'x');
+      await tester.pumpAndSettle(const Duration(seconds: 2));
+
+      final String editText =
+          _findRenderEditable(tester, passwordInputFinder).text.text;
```

Comment:
I think we should just check that the password field has the obscure property set to true instead of trying to test that it obscures the text.